code/modules/projectiles/projectile/special/meteor.dm

```diff
@@ -1,3 +1,120 @@
+#define DEFAULT_METEOR_LIFETIME 1800
+#define MAP_EDGE_PAD 5
+
+
+GLOBAL_VAR_INIT(meteor_wave_delay, 625) //minimum wait between waves in tenths of seconds
+//set to at least 100 unless you want evarr ruining every round
+
+//Meteors probability of spawning during a given wave
+GLOBAL_LIST_INIT(meteors_normal, list(/obj/projectile/meteor/dust=3, /obj/projectile/meteor/medium=8, /obj/projectile/meteor/big=3, \
+						  /obj/projectile/meteor/flaming=1, /obj/projectile/meteor/irradiated=3)) //for normal meteor event
+
+GLOBAL_LIST_INIT(meteors_threatening, list(/obj/projectile/meteor/medium=4, /obj/projectile/meteor/big=8, \
+						  /obj/projectile/meteor/flaming=3, /obj/projectile/meteor/irradiated=3)) //for threatening meteor event
+
+GLOBAL_LIST_INIT(meteors_catastrophic, list(/obj/projectile/meteor/medium=5, /obj/projectile/meteor/big=75, \
+						  /obj/projectile/meteor/flaming=10, /obj/projectile/meteor/irradiated=10, /obj/projectile/meteor/tunguska = 1)) //for catastrophic meteor event
+
+GLOBAL_LIST_INIT(meteorsB, list(/obj/projectile/meteor/meaty=5, /obj/projectile/meteor/meaty/xeno=1)) //for meaty ore event
+
+GLOBAL_LIST_INIT(meteorsC, list(/obj/projectile/meteor/dust)) //for space dust event
+
+GLOBAL_LIST_INIT(meteors_stress_test, list(/obj/projectile/meteor/tunguska))
+
+GLOBAL_LIST_EMPTY(meteor_circle)
+GLOBAL_LIST_EMPTY(meteor_circle_directions)
+
+
+///////////////////////////////
+//Meteor spawning global procs
+///////////////////////////////
+
+/proc/spawn_meteors(number = 10, list/meteortypes, direction)
```

Comment:
I think everything in here should either be in its own subsystem or in SSaugury.